### PR TITLE
Use DD_MASS_ACT_C3 for DDIC activation

### PR DESCRIPTION
**Pull request by christianguenter2**
#1055 
Use FM DD_MASS_ACT_C3 for DDIC activation if available. Must be enabled with experimental feature flag, otherwise the current activation logic is used.

### CODE REVIEW

File: src/zabapgit_objects.prog.abap [hunk 1]
```diff
@@ -155,6 +171,198 @@ CLASS lcl_objects_activation IMPLEMENTATION.
...
+  METHOD activate_new.
+
+    DATA: lt_ddic     TYPE act_work_area,
+          lt_non_ddic TYPE act_work_area.
+
+    IF gt_objects IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    zcl_abapgit_progress=>show( iv_key     = 'Activate'
```
Comment (larshp) [inline src/zabapgit_objects.prog.abap]: "Activate" -> "Activating DDIC"

File: src/zabapgit_objects.prog.abap [hunk 2]
```diff
@@ -155,6 +171,198 @@ CLASS lcl_objects_activation IMPLEMENTATION.
...
+      IF sy-subrc <> 0.
+        zcx_abapgit_exception=>raise( 'error from RS_WORKING_OBJECTS_ACTIVATE' ).
+      ENDIF.
+
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD filter_ddic_objects.
```
Comment (larshp) [inline src/zabapgit_objects.prog.abap]: do we have some objects in abapGit that identifies themselves as DDIC, but are not really DDIC? Or is there a difference between the function modules?

Comment (christianguenter2) [reply src/zabapgit_objects.prog.abap]: Oh, I see. We have already the distinction between DDIC and non-DDIC, so filter_ddic_objects is superfluous. Will change the code.

File: src/zabapgit_objects.prog.abap (hunk 1 above)
Comment (christianguenter2) [reply src/zabapgit_objects.prog.abap]: changed

**Comment by christianguenter2** [comment]
Changed the logic, so that the distinction between DDIC and non-DDIC of abapGit ist used

File: src/zabapgit_objects.prog.abap [hunk 3]
```diff
@@ -155,6 +165,161 @@ CLASS lcl_objects_activation IMPLEMENTATION.
...
+      activate_ddic( ).
+
+    ELSE.
+
+      zcl_abapgit_progress=>show( iv_key     = 'Activating non DDIC'
+                                  iv_current = '98'
+                                  iv_total   = '100'
+                                  iv_text    = '...' ).
+
+      activate_non_ddic( ).
```
Comment (larshp) [inline src/zabapgit_objects.prog.abap]: perhaps just call `activate_old( abap_false )`, then method `activate_non_ddic` can be deleted

Comment (christianguenter2) [reply src/zabapgit_objects.prog.abap]: good point, changed...

**Comment by christianguenter2** [comment]
first try displaying errors:
image

**Pull request merged**